core/src/test/java/org/apache/brooklyn/enricher/stock/SensorPropagatingEnricherTest.java

```diff
@@ -215,4 +213,43 @@ public void testEnricherSpecThrowsOnPropagatesAndPropagatesAllSet() throws Excep
             if (ise == null) throw e;
         }
     }
+
+    @Test
+    public void testPropagateToDynamicSensor() {
```

Comment:
Would be good to have a specific test for the actual fix, rather than just this more complicated example. Maybe something like:

    @Test
    public void testSensorPropagatedWhenMappingUsedSameNameButDifferentType() throws Exception {
        AttributeSensor<String> origSensor = Sensors.newSensor(String.class, "origSensor");
        AttributeSensor<Object> sourceSensorFromYaml = Sensors.newSensor(Object.class, "origSensor");
        AttributeSensor<Object> targetSensor = Sensors.newSensor(Object.class, "newSensor");
        app.addEnricher(Enrichers.builder()
                .propagating(ImmutableMap.of(sourceSensorFromYaml, targetSensor))
                .from(entity)
                .build());
        entity.sensors().set(origSensor, "myval");
        EntityTestUtils.assertAttributeEqualsEventually(app, targetSensor, "myval");
    }